internal/internal_workflow_client.go

```diff
@@ -877,6 +927,69 @@ func (wc *WorkflowClient) ResetWorkflowExecution(ctx context.Context, request *w
 	return resp, nil
 }
 
+func (wc *WorkflowClient) UpdateWorkflowWithOptions(
```

Comment:
This should be interceptable too. I think it's probably better to make this the intercepted method and `UpdateWorkflow` call it.

Reply:
Hmm.. I see we did the same with `QueryWorkflow`, that seems like a bug.
Let's leave as is for this PR and discuss with @cretz / @Quinn-With-Two-Ns.

Reply:
It is interceptable. This is the overload that the simpler form calls. But the interceptor is called for the most advanced of the calls. In some cases where it makes sense we deviate from the client interface names in the interceptor interface In this case. If we added a `SignalWorkflowWithOptions` it'd go through `ClientOutboundInterceptor.SignalWorkflow` too.

EDIT: Actually I just realized we forgot to make `QueryWorkflowWithOptions` go through the interceptor. We should fix that. It is the intention. I have opened https://github.com/temporalio/sdk-go/issues/991.

Reply:
Updated to funnel both Update variants into the outbound interceptor and added a test to the update branch in sdk-features.